src/com/owncloud/android/authentication/PinCheck.java

```diff
@@ -0,0 +1,30 @@
+package com.owncloud.android.authentication;
+
+import android.app.Activity;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+
+import com.owncloud.android.MainApp;
+
+public class PinCheck extends Activity {
+
+    private static Long timestamp = 0l;
+    private static Long lastStart = 0l;
+    private static int PINCODE_TIMEOUT = 10000;
+    
+    public static void setUnlockTimestamp() {
+        timestamp = System.currentTimeMillis();
+    }
+
+    public static boolean checkIfPinEntry(){
+        if ((System.currentTimeMillis() - timestamp) > PINCODE_TIMEOUT &&
+            (System.currentTimeMillis() - lastStart) > PINCODE_TIMEOUT){
+            SharedPreferences appPrefs = PreferenceManager.getDefaultSharedPreferences(MainApp.getAppContext());
```

Comment:
Whenever possible is better not using the Application Context; in this case, a parameter Context ctx in the method would do the trick; the callers are always Activities, they only need to pass 'this' as the value for the parameter.